network-mux/src/Network/Mux/Bearer/Socket.hs

```diff
@@ -44,34 +44,59 @@ import qualified Network.Mux.Time as Mx
 -- 'MuxError'.
 --
 socketAsMuxBearer
-  :: Tracer IO Mx.MuxTrace
+  :: Maybe DiffTime
+  -> Tracer IO Mx.MuxTrace
   -> Socket.Socket
   -> MuxBearer IO
-socketAsMuxBearer tracer sd =
+socketAsMuxBearer sduTimeout_m tracer sd =
       Mx.MuxBearer {
         Mx.read    = readSocket,
         Mx.write   = writeSocket,
         Mx.sduSize = 12288
       }
     where
+      hdrLenght = 8
+
       readSocket :: HasCallStack => IO (Mx.MuxSDU, Time)
       readSocket = do
           traceWith tracer $ Mx.MuxTraceRecvHeaderStart
-          hbuf <- recvLen' True 8 []
+
+          -- Wait for the first part of the header without any timeout
+          h0 <- recvAtMost False hdrLenght
```

Comment:
Should this be `recvAtMost True`? The call it's replacing used `waitingOnNxtHeader = True`. And if it's not true here, then it is always false in every call site.

Reply:
Yes, is should be true.